From bc62219f03edebf62b81e582584744a2ab675ca7 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Sat, 5 Mar 2022 20:23:44 +0100
Subject: [PATCH 080/106] Revert "block, bfq: record how many queues have
 pending requests in bfq_group"

This reverts commit 60db903cda44c610ae1c4f6091735e5fd54d92f6.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-cgroup.c  |  1 -
 block/bfq-iosched.c | 21 ---------------------
 block/bfq-iosched.h |  1 -
 3 files changed, 23 deletions(-)

diff --git a/block/bfq-cgroup.c b/block/bfq-cgroup.c
index 141de5c79..a6abbdb3a 100644
--- a/block/bfq-cgroup.c
+++ b/block/bfq-cgroup.c
@@ -557,7 +557,6 @@ static void bfq_pd_init(struct blkg_policy_data *pd)
 				   */
 	bfqg->bfqd = bfqd;
 	bfqg->active_entities = 0;
-	bfqg->num_entities_with_pending_reqs = 0;
 	bfqg->rq_pos_tree = RB_ROOT;
 }
 
diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index 23ebf4024..b27327824 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -933,16 +933,6 @@ void bfq_weights_tree_add(struct bfq_data *bfqd, struct bfq_queue *bfqq,
 inc_counter:
 	bfqq->weight_counter->num_active++;
 	bfqq->ref++;
-
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
-	if (!entity->in_groups_with_pending_reqs) {
-		struct bfq_group *bfqg =
-			container_of(entity->parent, struct bfq_group, entity);
-
-		entity->in_groups_with_pending_reqs = true;
-		bfqg->num_entities_with_pending_reqs++;
-	}
-#endif
 }
 
 /*
@@ -959,17 +949,6 @@ void __bfq_weights_tree_remove(struct bfq_data *bfqd,
 		return;
 
 	bfqq->weight_counter->num_active--;
-
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
-	if (bfqq->entity.in_groups_with_pending_reqs) {
-		struct bfq_group *bfqg = container_of(bfqq->entity.parent,
-				struct bfq_group, entity);
-
-		bfqq->entity.in_groups_with_pending_reqs = false;
-		bfqg->num_entities_with_pending_reqs--;
-	}
-#endif
-
 	if (bfqq->weight_counter->num_active > 0)
 		goto reset_entity_pointer;
 
diff --git a/block/bfq-iosched.h b/block/bfq-iosched.h
index 43f5bd857..ec447f052 100644
--- a/block/bfq-iosched.h
+++ b/block/bfq-iosched.h
@@ -949,7 +949,6 @@ struct bfq_group {
 	struct bfq_entity *my_entity;
 
 	int active_entities;
-	int num_entities_with_pending_reqs;
 
 	struct rb_root rq_pos_tree;
 
-- 
2.35.1.354.g715d08a9e5

